### PR TITLE
Add tests for AuthorizeUnknownUser

### PR DESCRIPTION
**Pull request by Igarfinkle**
## Description

Adds tests for AuthorizeUnknownUser

## Reviewer Notes

Isaac asks: do these tests represent a complete set of tests for this function?

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/167462990) for this change

### CODE REVIEW

**Comment by garrettqmartin8** [comment]
@lynzt that's a good point about covering cases for tsp and admin as well and that'd be good to see in this PR. As for splitting up the function, I think that may be out of scope for this story. My main goal with this was to make sure we had coverage that we could lean on if and when we decide to refactor this function.

File: pkg/models/admin_user.go [hunk 1]
```diff
@@ -106,3 +107,16 @@ func (a *AdminUser) ValidateCreate(tx *pop.Connection) (*validate.Errors, error)
 func (a *AdminUser) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
 	return validate.NewErrors(), nil
 }
+
+// FetchTspUserByEmail looks for an tsp user with a specific email
+func FetchAdminUserByEmail(tx *pop.Connection, email string) (*AdminUser, error) {
```
Comment (garrettqmartin8) [inline pkg/models/admin_user.go]: I think we're trying to move away from this pattern of defining ad-hoc query methods to each model in favor of the new query builder. Is the `AuthorizeUnknownUser` harder to test when using the query builder? If so, that'd would be good to know so we can make adjustments.

File: pkg/auth/authentication/auth.go [hunk 2]
```diff
@@ -538,11 +538,11 @@ func authorizeUnknownUser(openIDUser goth.User, h CallbackHandler, session *auth
 	if session.IsAdminApp() {
 		queryBuilder := query.NewQueryBuilder(h.db)
 		filters := []services.QueryFilter{
-			query.NewQueryFilter("email", "=", strings.ToLower(session.Email)),
+			query.NewQueryFilter("email", "=", session.Email),
 		}
 		err = queryBuilder.FetchOne(&adminUser, filters)
 
-		if err == models.ErrFetchNotFound {
+		if err != nil && errors.Cause(err).Error() == models.RecordNotFoundErrorString {
```
Comment (garrettqmartin8) [inline pkg/auth/authentication/auth.go]: Nice.

File: pkg/models/admin_user.go (hunk 1 above)
Comment (garrettqmartin8) [inline pkg/models/admin_user.go]: We can probably remove this now.

File: pkg/auth/authentication/auth_test.go [hunk 3]
```diff
@@ -571,3 +575,351 @@ func (suite *AuthSuite) TestAuthorizeDisableAdmin() {
...
+		false,
+	}
+	rr := httptest.NewRecorder()
+	span := trace.Span{}
+
+	authorizeUnknownUser(user, h, &session, rr, &span, req.WithContext(ctx), "")
+
+	// Office app, so should only have office ID information
+	suite.Equal(officeUser.ID, session.OfficeUserID)
+	suite.Equal(uuid.Nil, session.TspUserID)
```
Comment (lynzt) [inline pkg/auth/authentication/auth_test.go]: add: `suite.Equal(uuid.Nil, session.AdminUserId)`

File: pkg/auth/authentication/auth_test.go [hunk 4]
```diff
@@ -571,3 +575,351 @@ func (suite *AuthSuite) TestAuthorizeDisableAdmin() {
...
+		false,
+	}
+	rr := httptest.NewRecorder()
+	span := trace.Span{}
+
+	authorizeUnknownUser(user, h, &session, rr, &span, req.WithContext(ctx), "")
+
+	// Office app, so should only have office ID information
+	suite.Equal(tspUser.ID, session.TspUserID)
+	suite.Equal(uuid.Nil, session.OfficeUserID)
```
Comment (lynzt) [inline pkg/auth/authentication/auth_test.go]: add: suite.Equal(uuid.Nil, session.AdminUserId)

File: pkg/auth/authentication/auth_test.go [hunk 5]
```diff
@@ -571,3 +575,351 @@ func (suite *AuthSuite) TestAuthorizeDisableAdmin() {
...
+		false,
+	}
+	rr := httptest.NewRecorder()
+	span := trace.Span{}
+
+	authorizeUnknownUser(user, h, &session, rr, &span, req.WithContext(ctx), "")
+
+	// Office app, so should only have office ID information
+	suite.Equal(adminUser.ID, session.AdminUserID)
+	suite.Equal(uuid.Nil, session.OfficeUserID)
```
Comment (lynzt) [inline pkg/auth/authentication/auth_test.go]: add: `suite.Equal(uuid.Nil, session.TspUserId)`

**Pull request merged**